### PR TITLE
Unit tests for core ML functionality

### PR DESCRIPTION
**Pull request by breville**
Two unit tests for the core ML functionality: one for numerical data, and one for categorical.

### CODE REVIEW

File: test/unit/train.test.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    train.init(store);
+    train.onClickTrain(store);
+
+    store.dispatch(setTestData({temperature: "150", rain: "1030"}));
+
+    train.onClickPredict(store);
+
+    const predictedValue = getConvertedPredictedLabel(store.getState());
+    expect(predictedValue).toBe(30);
```
Comment (Erin007) [inline test/unit/train.test.js:43]: Hmmm. Looking at the training data, I'd expect the predictedValue to be 35 since the most similar training data point where temp is 150 and rain is 1030 is 35, not 30.  However, I do think it's valuable to have coverage while we figure out how to handle this situation, and getting this test set up merged in will allow us to iterate more quickly. Could we maybe include a comment about how if the K value optimization goes sideways with really small datasets such as this sample one?

File: test/unit/train.test.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    train.init(store);
+    train.onClickTrain(store);
+
+    store.dispatch(setTestData({flavor: "sweet", texture: "crunchy"}));
+
+    train.onClickPredict(store);
+
+    const predictedLabel = getConvertedPredictedLabel(store.getState());
+
+    expect(predictedLabel).toBe("yellow");
```
Comment (Erin007) [inline test/unit/train.test.js:77]: similar to above, given what we know training is currently doing, "yellow" makes sense but we'd like to tweak things so that the expected is actually "green".

**Review by Erin007 — APPROVED** [review]
I'm ok with this going in as is, but would as we muddle through the accuracy wonkiness I'd like to expand the test suite to include more variety - large datasets, small datasets, datasets with all unique feature values and labels, datasets with duplicates etc There's some complexity buried in here that I don't want us to gloss over.

File: test/unit/train.test.js (hunk 1 above)
Comment (Erin007) [reply test/unit/train.test.js:43]: Oh, actually this makes sense, because the model is never trained on ` { temperature: "150", cost: "35", rain: "1030" }` because we pull that row to use as test data to calculate accuracy.

Comment (Erin007) [reply test/unit/train.test.js:43]: Is there a way to set up these tests so we pass the training data rather than the whole dataset?

**Comment by Erin007** [comment]
@breville how would you feel about merging #234 into this branch?

File: src/train.js [hunk 3]
```diff
@@ -74,16 +72,18 @@ const extractTrainingLabel = (state, row) => {
...
+    extractTrainingExamples(updatedState, row)
+  );
+  const trainingLabels = updatedState.data.map(row =>
+    extractTrainingLabel(updatedState, row)
+  );
   /*
   Select X% of examples and corresponding labels from the training set to use for a post-training accuracy calculation.
   */
-  const numToReserve = parseInt(
+  const numToReserve = Math.ceil(
```
Comment (breville) [inline src/train.js:86]: @Erin007 Actually, will we always want to round up?

File: test/unit/train.test.js (hunk 1 above)
Comment (breville) [reply test/unit/train.test.js:43]: Good question.  This is something of an architecture question, and comes down to boundaries between modules.  I chose to unit test the exports from `train.js`.  We could unit test at a deeper level, or break apart some of the steps encapsulated in `init`, but I think we should do that refactoring separately.  This is a good example of how API designs can be informed by unit testing, and vice versa.

File: src/train.js (hunk 3 above)
Comment (Erin007) [reply src/train.js:86]: I can't think of a time when we wouldn't want to round up.
